### PR TITLE
Use store for quizes.

### LINKED ISSUE #2: Main page
Main page flow (desktop & mobile): 

- Create store for saving data information.
- Fix url for quiz page.

### PR DESCRIPTION
**Pull request by DGMary**

### CODE REVIEW

File: src/components/Quiz/QuizItems.tsx [hunk 1]
```diff
@@ -1,30 +1,54 @@
...
+  width: '100%'
+}));
 
 const QuizItems = () => {
-  const [quizInfo, setQuizInfo] = useState([]);
-  const quizesCollectionRef = collection(db, 'quizes');
+  const { quizes, status, error } = useAppSelector((state) => state.quizesReducer);
+  const dispatch = useAppDispatch();
 
   useEffect(() => {
```
Comment (DGMary) [inline src/components/Quiz/QuizItems.tsx:29]: 24-30 row make useCallback

File: src/components/Quiz/QuizItems.tsx [hunk 2]
```diff
@@ -1,30 +1,54 @@
...
   }, []);
 
   return (
     <Grid container spacing={2} sx={{ alignItems: 'stretch', paddingTop: 2, paddingBottom: 4 }}>
-      {quizInfo.map((item) => (
-        <Grid item xs={12} sm={6} md={4} key={item.id}>
-          <QuizItem item={item} />
-        </Grid>
-      ))}
+      {error !== 'Network Error' && status !== 'failed' && (
```
Comment (DGMary) [inline src/components/Quiz/QuizItems.tsx]: Save checking to variable.

File: src/components/Quiz/QuizItems.tsx [hunk 3]
```diff
@@ -1,30 +1,54 @@
...
-    getQuizes();
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    (async () => {
+      try {
+        await dispatch(quizesThunks.fetchQuizes());
+      } catch (e) {
+        console.error(e);
+      }
+    })();
   }, []);
```
Comment (kusochek) [inline src/components/Quiz/QuizItems.tsx:37]:
```
if (loading) return  <CircularProgressWrapper>...
if (error) return <ErrorWrapper>....
return (
      <>
      {quizes.map((item) => (
        <Grid item xs={12} sm={6} md={4} key={item.id}>
          <QuizItem item={item} />
        </Grid>
      ))}
      </>
  )
```

File: src/store/services/quizes/index.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export { default as quizesReducer } from './reducer';
+export { default as quizesThunks } from './thunks';
+// export { default as quizesActions } from './actions';
```
Comment (DGMary) [inline src/store/services/quizes/index.ts]: Add TODO: "description" delete if dosn`t need this logic

File: src/store/services/quizes/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { moduleName } from './constant';
+import thunks from './thunks';
+import { IQuizItemProps } from '../../../types/quiz';
+
+interface IInitialState {
+  quizes: IQuizItemProps[];
+  status: 'idle' | 'loading' | 'success' | 'failed';
```
Comment (DGMary) [inline src/store/services/quizes/reducer.ts]: make status into enum type

File: src/store/services/quizes/reducer.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-nocheck
```
Comment (DGMary) [inline src/store/services/quizes/reducer.ts]: Make check types for file.

File: src/components/Quiz/QuizItems.tsx [hunk 7]
```diff
@@ -1,30 +1,53 @@
...
-      ))}
+      {isNetworkOk && (
+        <>
+          {status === 'loading' && <ProgressWrapper><CircularProgress color="inherit"/></ProgressWrapper>}
+          {status === 'success' && <>{quizesList}</>}
+        </>
+      )}
     </Grid>
   );
 };
```
Comment (kusochek) [inline src/components/Quiz/QuizItems.tsx]: it should be like that

```
if (isNetworkOk && status === 'loading) return <ProgressWrapper><CircularProgress color="inherit"/></ProgressWrapper>
if (!success) return <SomeErrorComponent />

return (
    <Grid container spacing={2} sx={{ alignItems: 'stretch', paddingTop: 2, paddingBottom: 4 }}>
          {quizes.map((item) => (
    <Grid item xs={12} sm={6} md={4} key={item.id}>
      <QuizItem item={item} />
    </Grid>}
      )}
    </Grid>
  )
  ```

**Review by kusochek — APPROVED** [review]

**Pull request merged**